### ISSUE TITLE
roomseg issue

### ISSUE THREAD

**Issue by hzh3gg** [PRBonn/SIMP#10]
May I ask how to generate the roomseg.png image in mapping_dode when running online

**Comment by FullMetalNicky** [comment]
We already explained it in one of the closed issues...please check them.

**Comment by hzh3gg** [comment]
Sorry to bother you, this is a great project, the args.json file has "gmap", "roomseg", "categories", "yolo_conf", "omni_conf", "min_dist", "max_dist", "width", "height", "2DvisTest", "3DvisTest", "YOLOvisTest", "odomTrigger", "odom_trigerTH", "totalSeenTH", "costTH", "truncTH", "skipTH", "variancePath", "K", "cam_poses" ", "angles", "min_z", "camera_z" parameters, when running mapping_node in online mode, which parameters need to be modified according to the actual situation?

**Comment by hzh3gg** [comment]
Sorry to bother you, this is a great project. There are 24 parameters in the args.json file such as "gmap", "roomseg", "categories", etc. When running mapping_node in online mode, which parameters need to be modified according to the actual situation? Where is

**Comment by hzh3gg** [comment]
Thank you for your reply. One more question, was your well-trained model trained on your own dataset collected for the scenario in your laboratory？



---Original---
From: ***@***.***&gt;
Date: Wed, Feb 28, 2024 18:32 PM
To: ***@***.***&gt;;
Cc: "Zhihao ***@***.******@***.***&gt;;
Subject: Re: [PRBonn/SIMP] roomseg issue (Issue #10)




 
Closed #10 as completed.

**Comment by FullMetalNicky** [comment]
The model was fine-tuned on objects in one floor of the lab, and then tested also in the other floor. 
We needed specific classes that were not part of the trained model provided by Meta.

**Comment by FullMetalNicky** [comment]
The names in the args,json are pretty self explanatory. If you are in a different building, with a different floor plan then obviously you need to give the right floor plan as input to the algorithm. If your model is trained to detect different categories than ours, you'll have to change the categories. This also implies computing the variance and TH for this classes/model...
If you use a different camera, then you need to provide your own calibration and image dimensions...

**Comment by hzh3gg** [comment]
Thanks, I have a few questions about camera pose configuration:

I didn't find it in previous comments. Camera_pose is the position of the camera relative to which coordinate system. Is it relative to base_link?
Because I'm only using 1 camera (realsense d435i), located about 0.39m above the ground, 0.15m x-axis away from base_link and published manually:
<node name="base_link_to_camera" pkg="tf" type="static_transform_publisher" args="0.15 0 0.39 -1.5707963 0 -1.5707963">
So I only configured the first camera and I
    "cam_poses" : [0.15, 0, 0.39, 0, .........],
    "angles": [0,..........],
    "min_z": 0,
    "camera_z": 0.39
Are these four parameters configured correctly? If not, how should they be configured correctly?

**Comment by hzh3gg** [comment]
Thanks, I have a few questions about camera pose configuration:

I didn't find it in previous comments. Camera_pose is the position of the camera relative to which coordinate system. Is it relative to base_link?
Because I'm only using 1 camera (realsense d435i), located about 0.39m above the ground, 0.15m x-axis away from base_link and published manually:

So I only configured the first camera:
"cam_poses" : [0.15, 0, 0.39, 0, .........],
"angles": [0,..........],
"min_z": 0,
"camera_z": 0.39
Are these four parameters configured correctly? If not, how should they be configured correctly?

**Comment by hzh3gg** [comment]
Thanks, I have a few questions about camera pose configuration:

I didn't find it in previous comments. Camera_pose is the position of the camera relative to which coordinate system. Is it relative to base_link?
Because I'm only using 1 camera (realsense d435i), located about 0.39m above the ground, 0.15m x-axis away from base_link and published manually:
<node name="base_link_to_camera" pkg="tf" type="static_transform_publisher" args="0.15 0 0.39 -1.5707963 0 -1.5707963">
So I only configured the first camera:
    "cam_poses" : [0.15, 0, 0.39, 0, .........],
    "angles": [0,..........],
    "min_z": 0,
    "camera_z": 0.39
Are these four parameters configured correctly? If not, how should they be configured correctly?

**Comment by hzh3gg** [comment]
Thanks, I have a few questions about camera pose configuration:

I didn't find it in previous comments. Camera_pose is the position of the camera relative to which coordinate system. Is it relative to base_link?
Because I'm only using 1 camera (realsense d435i), located about 0.39m above the ground, 0.15m x-axis away from base_link and published manually:
`<node name="base_link_to_camera" pkg="tf" type="static_transform_publisher" args="0.15 0 0.39 -1.5707963 0 -1.5707963">`
So I only configured the first camera:
    "cam_poses" : [0.15, 0, 0.39, 0, .........],
    "angles": [0,..........],
    "min_z": 0,
    "camera_z": 0.39
Are these four parameters configured correctly? If not, how should they be configured correctly?

**Comment by hzh3gg** [comment]
Thanks, I have a few questions about camera pose configuration:

I didn't find it in previous comments. Camera_pose is the position of the camera relative to which coordinate system. Is it relative to base_link?
Because I'm only using 1 camera (realsense d435i), located about 0.39m above the ground, 0.15m x-axis away from base_link and published manually:
`<node name="base_link_to_camera" pkg="tf" type="static_transform_publisher" args="0.15 0 0.39 -1.5707963 0 -1.5707963">`
So I only configured the first camera:
```
"cam_poses" : [0.15, 0, 0.39, 0, .........],
"angles": [0,..........],
"min_z": 0,
"camera_z": 0.39
```
Are these four parameters configured correctly? If not, how should they be configured correctly?

**Comment by hzh3gg** [comment]
Thanks, I have a few questions about camera pose configuration:

I didn't find it in previous comments. Camera_pose is the position of the camera relative to which coordinate system. Is it relative to base_link?
Because I'm only using 1 camera (realsense d435i), located about 0.39m above the ground, 0.15m x-axis away from base_link and published manually:
<node name="base_link_to_camera" pkg="tf" type="static_transform_publisher" args="0.15 0 0.39 -1.5707963 0 -1.5707963"&gt;
So I only configured the first camera:
"cam_poses" : [0.15, 0, 0.39, 0, .........], "angles": [0,..........], "min_z": 0, "camera_z": 0.39 


Are these four parameters configured correctly? If not, how should they be configured correctly?





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PRBonn/SIMP"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月8日(星期五) 凌晨0:29
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PRBonn/SIMP] roomseg issue (Issue #10)





 
The names in the args,json are pretty self explanatory. If you are in a different building, with a different floor plan then obviously you need to give the right floor plan as input to the algorithm. If your model is trained to detect different categories than ours, you'll have to change the categories. This also implies computing the variance and TH for this classes/model...
 If you use a different camera, then you need to provide your own calibration and image dimensions...

**Comment by hzh3gg** [comment]
Sorry to bother you again,
I used the weights you retrained for the 3D object detection model, but I could hardly detect any objects correctly, so I used the original model, I would like to ask how should the variance.csv file corresponding to the original model be generated, or can I Where to find the variance.csv file corresponding to the original model?
Sorry to bother you again, thank you very much




---Original---
From: ***@***.***&gt;
Date: Fri, Mar 8, 2024 00:29 AM
To: ***@***.***&gt;;
Cc: "Zhihao ***@***.******@***.***&gt;;
Subject: Re: [PRBonn/SIMP] roomseg issue (Issue #10)




 
The names in the args,json are pretty self explanatory. If you are in a different building, with a different floor plan then obviously you need to give the right floor plan as input to the algorithm. If your model is trained to detect different categories than ours, you'll have to change the categories. This also implies computing the variance and TH for this classes/model...
 If you use a different camera, then you need to provide your own calibration and image dimensions...

**Comment by FullMetalNicky** [comment]
Our model, as we explained in another closed issue, was fine-tuned for the specific PoV of our robot. 
If your cameras are mounted higher, then you can expect the original model to perform better. 
For computing the variances, please read the paper.
